### PR TITLE
bug-fix: misc changes for MacOS Sonoma compatibility

### PR DESCRIPTION
**Pull request by miklhh**
This commit contains the changes I needed to compile apytypes on my MacOS Sonoma (x86-64) machine.

Note: On some target architectures, the token `OVERFLOW` is reserved in `math.h` and cannot be used throughout source code. My MacOS machine was such an example where this caused trouble. I renamed the fixed-point overflow mode `OVERFLOW` to `TWOS_OVERFLOW`. Other name suggestions are welcome.

### CODE REVIEW

**Comment by oscargus** [comment]
Why is .clangd required? Also, the Makefile does not serve any purpose long term. Hence, I suggest get rid of that and make sure that it compiles correctly when installed.

File: src/.gitignore [hunk 1]
```diff
@@ -9,6 +9,10 @@ callgrind.out.*
 # Apple debuging symbol files
 *.dSYM
 
+# .clangd, as this is host machine specific. See .clangd-gen.py to genererate a new
+# .clangd configuration file instead
+.clangd
```
Comment (oscargus) [inline src/.gitignore:14]: Use a single, global, .gitignore

**Comment by miklhh** [comment]
> Why is .clangd required? Also, the Makefile does not serve any purpose long term. Hence, I suggest get rid of that and make sure that it compiles correctly when installed.

.clangd is the configuration file for Clangd, the LSP we use when writing the C++. I have opened an issue, #8, and will get around to migrating the tests to Meson as soon as I'm done with the MacOS ci.

File: src/.gitignore (hunk 1 above)
Comment (miklhh) [reply src/.gitignore]: Fine, I have moved the content of `src/.gitignore` to `.gitignore`

**Comment by oscargus** [comment]
I do not really see the point of having a .clangd-file in the repository. Especially not one with absolute paths to specific Python-versions. It should be fully feasible to have one locally for those using this.

**Comment by miklhh** [comment]
That is why I removed the .clangd file, added it to the .gitignore, and replaced it with a script that can automate the process of generating a new one.

**Comment by miklhh** [comment]
Or were referring to the `.clangd-gen.py` generation file? I could see that it might me a bit redundant too, although I believe that clangd is the by far most popular LSP for C/C++. Do you want me to remove it?

**Pull request merged**